ambry-api/src/main/java/com/github/ambry/config/StoreConfig.java

```diff
@@ -319,6 +319,29 @@ public class StoreConfig {
   public static final String storeReadWriteEnableSizeThresholdPercentageDeltaName =
       "store.read.write.enable.size.threshold.percentage.delta";
 
+  /**
+   * Specifies the size threshold (as percentage of maximum size) of a store for converting the replica and partition to
+   * partially writable from RW.
+   * A partially writable replica/partition will not be able to take writes for metadata chunks whose id is reserved at
+   * the start of the upload.
+   */
+  @Config(storePartialWriteEnableSizeThresholdPercentageName)
+  @Default("50")
+  public final int storePartialWriteEnableSizeThresholdPercentage;
+  public static final String storePartialWriteEnableSizeThresholdPercentageName =
+      "store.partial.write.enable.size.threshold.percentage";
+
+  /**
+   * Specifies the size threshold delta below {@link #storePartialWriteEnableSizeThresholdPercentageName} that a store will
+   * be converted from partially writable to RW
+   */
+  @Config(storePartialWriteEnableSizeThresholdPercentageDeltaName)
+  @Default("5")
+  public final int storePartialWriteEnableSizeThresholdPercentageDelta;
```

Comment:
Since this is a config related to converting partially writable to RW, would it be better if we name it as `storePartialWriteToReadWriteEnableSizeThresholdPercentageDelta`